### PR TITLE
[bansearch] format, redir, event 352, 354

### PR DESCRIPTION
**Pull request by bw1**
@nhandler

### CODE REVIEW

File: scripts/bansearch.pl [hunk 1]
```diff
@@ -218,7 +227,21 @@ sub RPL_WHOREPLY {
 
         return if(!$running);
 
-        (undef, $user, $host, $nick, $account, $real) = split(/ /, $data, 6);
+	# 0   1 2   3       4               5    6 7  8
+	# bw2 * ~pi rpi1.my irc.example.net rpi1 H :0 pi
```
Comment (ailin-nemui) [inline scripts/bansearch.pl]: where did you see this response?

I think it is wrong

$ownnick, $channel_or_star, $user, $host, $server, $nick, $here_or_gone, $hopcount, $real

it's H/G for away status, not account

also, you are missing the split limit (if real name contains spaces)

File: scripts/bansearch.pl [hunk 2]
```diff
@@ -218,7 +227,21 @@ sub RPL_WHOREPLY {
...
+}
+
+sub RPL_WHOSPCRPL {
+	my($server, $data) = @_;
+
+        return if(!$running);
+
+	# 0   1    2               3   4   5
+	# bw2 ~bw1 irc.example.net bw1 bw2 :bw1
+        (undef, $user, $host, $nick, $account, $real) = split(/ /, $data);
```
Comment (ailin-nemui) [inline scripts/bansearch.pl]: split limit, if real name has spaces

File: scripts/bansearch.pl [hunk 3]
```diff
@@ -51,7 +51,7 @@ sub bansearch {
 	Irssi::active_win()->print("\x02Channel\x02: $channel");
 
         #Perform a /who <user> %uhnar
-	$server->redirect_event('who',1, $person, 0, undef,
```
Comment (ailin-nemui) [inline scripts/bansearch.pl]: (just a note; the matcher ($person) only works for channels)

**Comment by bw1** [comment]
thx

**Pull request merged**